dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/tracker/export/fields/InstanceFieldsParamMapper.java

```diff
@@ -38,17 +38,17 @@
 import org.hisp.dhis.fieldfiltering.FieldPath;
 
 /**
- * Abstract class that provides basic methods to transform input fields into
- * {@link FieldPath } based on {@link FieldFilterParser }. It follows the
- * principles of {@link org.hisp.dhis.fieldfiltering.FieldFilterService}
+ * Provides basic methods to transform input fields into {@link FieldPath }
+ * based on {@link FieldFilterParser }. It follows the principles of
+ * {@link org.hisp.dhis.fieldfiltering.FieldFilterService}
  */
-abstract class InstanceFieldsSupportService
+public interface InstanceFieldsParamMapper
```

Comment:
I don't think we need an interface. The mapping functions in for example TrackerTrackedEntityFieldsParamMapper are just pure functions. We don't need to mock these in tests of the TrackerTrackedEntityController. We also just have one implementation for each type of controller. So IMHO the reasons I look to when creating an interface are not given here. All that we need is for functions like TrackerTrackedEntityFieldsParamMapper.map to be able to reuse some common functions that are in here 😄 

I assume the prefix `InstanceFieldsParamMapper` comes from `TrackedEntityInstance`? If so that does not apply here as this class only knows about field paths. Prefix in `InstanceParams` is probably also not accurate 🤔

Reply:
In commit [999f705](https://github.com/dhis2/dhis2-core/pull/11837/commits/999f705ce044507edcd8a8d876a3d956966f8a23) i've already moved it to a class. 
I've renamed it to `FieldsPathParamsHelper`